akka-stream-tests/src/test/scala/akka/stream/io/ByteStringParserSpec.scala

```diff
@@ -0,0 +1,50 @@
+package akka.stream.io
+
+import akka.stream.{ ActorMaterializer, Attributes, ThrottleMode }
+import akka.stream.impl.io.ByteStringParser
+import akka.stream.scaladsl.{ Sink, Source }
+import akka.stream.stage.GraphStageLogic
+import akka.stream.testkit.StreamSpec
+import akka.util.ByteString
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class ByteStringParserSpec extends StreamSpec() {
+  implicit val materializer = ActorMaterializer()
+
+  class Chunker extends ByteStringParser[ByteString] {
+    import ByteStringParser._
+
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new ParsingLogic {
+
+      lazy val step: ParseStep[ByteString] = new ParseStep[ByteString] {
+        override def parse(reader: ByteReader): ParseResult[ByteString] = {
+          val bytes = reader.take(2)
+          ParseResult(Some(bytes), step)
+        }
+      }
+
+      startWith(step)
+
+    }
+
+  }
+
+  "ByteStringParser" must {
+
+    "respect backpressure" in {
```

Comment:
Maybe add a comment that explains how that stream setup tests that backpressure works correctly.